src/jit/rationalize.cpp

```diff
@@ -116,6 +116,16 @@ void Rationalizer::RewriteSIMDOperand(LIR::Use& use, bool keepBlk)
         addr->gtType = simdType;
         use.ReplaceWith(comp, addr);
     }
+#if defined(_TARGET_X86_)
+    // For x86, we need to handle GT_IND(GT_ADDR(GT_SIMD))
+    else if ((addr->OperGet() == GT_ADDR) && (addr->gtGetOp1()->OperGet() == GT_SIMD))
+    {
+        BlockRange().Remove(tree);
+        BlockRange().Remove(addr);
+
+        use.ReplaceWith(comp, addr->gtGetOp1());
+    }
+#endif // defined(_TARGET_X86_)
```

Comment:
I wonder, why this logic is not needed for amd64.  I suspect either of the following
1) some logic in morph is handling this or
2) It is needed and just we haven't hit it on amd64?

If it is (1), perhaps we should examine to see if we need to enable that morph logic for x86 RyuJIT as well.  
It would be a good idea to see what happens when the same SIMD test is run on Amd64, how IR is getting transformed and why it is not hitting any asserts.